tests/integration/test_integration.py

```diff
@@ -898,3 +896,48 @@ def test_inin022_no_callback_context():
     for attr in ['inputs', 'states', 'triggered', 'response']:
         with pytest.raises(MissingCallbackContextException):
             getattr(callback_context, attr)
+
+
+def test_inin023_wrong_callback_id():
+    app = Dash(__name__)
+    app.layout = html.Div([
+        html.Div([
+            html.Div(id='inner-div'),
+            dcc.Input(id='inner-input')
+        ], id='outer-div'),
+        dcc.Input(id='outer-input')
+    ], id='main')
+
+    ids = ['main', 'inner-div', 'inner-input', 'outer-div', 'outer-input']
+
+    with pytest.raises(NonExistentIdException) as err:
+        @app.callback(
+            Output('nuh-uh', 'children'),
```

Comment:
might be slightly better with a multi-output edge case?

Reply:
added multi-output in https://github.com/plotly/dash/pull/821/commits/fbec9dcd866a847e6299d3130bbf8d8cdfce8619